dotcom-rendering/src/web/components/TableOfContents.importable.tsx

```diff
@@ -1,72 +1,98 @@
 import { css } from '@emotion/react';
-import { line, neutral, space, textSans } from '@guardian/source-foundations';
+import { ArticleDisplay } from '@guardian/libs';
+import { headline, line, space, textSans } from '@guardian/source-foundations';
 import {
 	SvgChevronDownSingle,
 	SvgChevronUpSingle,
 } from '@guardian/source-react-components';
 import { useState } from 'react';
 import type { TableOfContentsItem } from '../../types/frontend';
+import type { Palette } from '../../types/palette';
+import { decidePalette } from '../lib/decidePalette';
 
 interface Props {
 	tableOfContents: TableOfContentsItem[];
+	format: ArticleFormat;
 }
 
-const anchorStyles = css`
-	color: black;
+const anchorStyles = (palette: Palette) => css`
+	color: ${palette.text.tableOfContents};
 	text-decoration: none;
+	display: block;
+	padding: ${space[1]}px 0 ${space[4]}px 0;
 `;
 
-const listItemStyles = css`
-	${textSans.xsmall({ fontWeight: 'bold', lineHeight: 'regular' })}
-	border-bottom: 1px solid ${line.primary};
-	padding: 6px 0;
-
-	&:last-child {
-		border-bottom: none;
+const defaultListItemStyles = (palette: Palette) => css`
+	border-top: 1px solid ${line.primary};
+	&:hover {
+		border-top: 1px solid ${palette.text.tableOfContents};
+		cursor: pointer;
 	}
 `;
 
+const listItemStyles = (format: ArticleFormat, palette: Palette) => {
+	if (format.display === ArticleDisplay.Immersive) {
+		return css`
+			${headline.xxxsmall({ fontWeight: 'light' })}
+			${defaultListItemStyles(palette)}
+		`;
+	}
+
+	return css`
+		${headline.xxxsmall({ fontWeight: 'bold' })}
+		${defaultListItemStyles(palette)}
+	`;
+};
```

Comment:
If you wanted to remove the need of abstracting the default styles (reducing code and complexity?), you could:

```TS
const listItemStyles = (format: ArticleFormat, palette: Palette) => {
  const fontWeight =
    format.display === ArticleDisplay.Immersive ? 'light' : 'bold';
  
  return css`
    ${headline.xxxsmall({ fontWeight })};
    border-top: 1px solid ${line.primary};
    &:hover {
      border-top: 1px solid ${palette.text.tableOfContents};
      cursor: pointer;
    }
  `;
};

```

It's still readable as is, just popped out to me.